defaults/main.yml

```diff
@@ -4,12 +4,13 @@
 # container image settings
 #
 
-awx_awxweb_version: 10.0.0
-awx_awxtask_version: 10.0.0
+awx_awx_version: 13.0.0
+awx_awxweb_version: "{{ awx_awx_version }}"
+awx_awxtask_version: "{{ awx_awx_version }}"
 awx_postgres_version: 10
 
-awx_awxweb_image: docker.io/ansible/awx_web:{{ awx_awxweb_version }}
-awx_awxtask_image: docker.io/ansible/awx_task:{{ awx_awxtask_version }}
+awx_awxweb_image: "{{ awx_awx_image | default('docker.io/ansible/awx:{}'.format(awx_awxweb_version)) }}"
```

Comment:
Please help me understand what's the purpose to use so complicated notation with .default in a file that is supposed to have just defaults? what's the difference if you'd just have {{ awx_awx_image }}: {{ awx_awxweb_version }} ?

This syntax was new to me, glad to learn it. I just don't get what does it add here?

Reply:
When providing just a default for awx_awx_image you are loosing the previous notation. I wanted to keep backward compatibility.

If awx_awx_image is not defined one can use:
* ``awx_awxweb_{image, version}``
* ``awx_awxtask_{image, version}``

in playbooks as before

or

one can define just ``awx_awx_image`` to set both images, e.g.

``awx_awx_image: quay.io/cfelder/awx:nopreload-{{ awx_awx_version }}``.

Reply:
ok, thanks